### ISSUE TITLE
Try out docker image shrinking with ReproZip

### ISSUE THREAD

**Issue by remram44** [ViDA-NYU/reprozip#141]
There is some interest in shrinking Docker images, which usually contain more stuff that is needed for the one task they are to perform, since they are created from a full operating system image. For example here: [blog post](http://blog.librato.com/posts/docker-images) ([tool](https://github.com/djosephsen/skinnywhale)).

ReproZip could totally do that, it just needs some wrapper code to interact with the Docker image (inject itself into it, get pack out?)

**Comment by satra** [comment]
@remram44 - i'm trying to run reprozip inside a docker container, connected to it in interactive mode. but even doing something like this `reprozip trace cat README.md` appears to take for ever.

i have reprozip installed using the vida-nyu conda channel inside a docker container.

**Comment by remram44** [comment]
Yes, I ran into this recently! The problem is that Docker has seccomp rules preventing the use of ptrace inside the container (among other "unusual" system calls).

Because of a tragic oversight, ReproZip doesn't print a message when that happens (fixed by 8d131ed6 which will be in 1.0.7), but the program doesn't execute.

Using the Docker option `--security-opt seccomp:unconfined` should fix this (or if security is an issue, write your own seccomp profile, making sure to enable ptrace).

See also [the Docker documentation on seccomp](https://github.com/docker/docker/blob/master/docs/security/seccomp.md).

**Comment by satra** [comment]
great - that worked.

**Comment by chrisfilo** [comment]
I just run into this. It would be great if reprozip would give a meaningful (or googlable) error instead of just hanging.

**Comment by remram44** [comment]
I released reprozip 1.0.7 which correctly display the message, and links to [this new troubleshooting entry](http://reprozip.readthedocs.io/en/1.0.x/troubleshooting.html#ptrace).

**Comment by remram44** [comment]
@kaczmarj seems to be doing just this in https://github.com/kaczmarj/neurodocker

**Comment by satra** [comment]
@remram44 - yes, he is working with us :)
